mantidproject/mantid
# Fix CCache compiler flags on Windows on initial cmake run

**Description of work.**

There was a dependency problem between CommonSetup.cmake and WindowsSetup.cmake where `include(CCacheSetup)` was run after the logic that tried to use it. This was causing the `/Zi` compiler flag to be used which effectively turns off CCache and slow down builds using Ninja on Windows. The problem was resolved by running cmake a second time so only an issue on a clean build but worth fixing anyway.

Resolve by running the include(CCacheSetup) command twice on Windows. Better than trying to fix an order between CommonSetup and WindowsSetup because there is some platform specific logic in CommonSetup eg if a dependency exists on all platforms but needs slightly different treatment on Windows vs Linux vs Mac.

**To test:**

Delete CMakeCache.txt
Open up "x64 Native Tools Visual Studio Command Prompt for VS"
Run cmake using ninja generator: `cmake --preset=win-ninja -B=build`
Check build.ninja file contains commands that use `/Z7` and not `/Zi`
Build code: `cd build` then `ninja`
Build code again - should be much quicker because ccache active

*There is no associated issue.*

*This does not require release notes* because **it affects developers only**



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
